keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,832 @@
 
-# Workflows
-Here we describe the workflows used to automate provisioning of new and existing buckets, and the de-provisioning of these buckets.
+In case of IAM style authentication, along with the `credentialsSecretName`, `serviceAccountName` field must also be specified. This will map the specified serviceaccount to the appropriate service account in the OSP.
 
-When the app pod is started, the kubelet will gRPC call _NodeStageVolume_ and _NodePublishVolume_, which are received by the cosi-node-adapter. The pod waits until the adapter responds to the gRPC requests. The adapter ensures that the target bucket has been provisioned and is ready to be accessed.
+```
+    BucketAccess - ba1                                        BucketAccessClass - bac2
+    |---------------------------------------|                 |----------------------------------|
+    | metadata:                             |                 | provisioner: s3.amazonaws.com    |
+    |   namespace: ns1                      |                 | parameters:                      |
+    | spec:                                 |                 |   key: value                     |
+    |   bucketAccessClassName: bac2         |                 | authenticationType: IAM          |
+    |   bucketClaimName: bcl1               |                 |----------------------------------|
+    |   credentialsSecretName: bucketcreds1 |
+    |   serviceAccountName: svacc1          |
+    | status:                               |
+    |   conditions:                         |
+    |   - name: AccessGranted               |
+    |     value: False                      |
+    |---------------------------------------|
+```
 
-When the app pod terminates, the kubelet gRPC calls _NodeUnstageVolume_ and _NodeUnpublishVolume_, which the cosi-node-adapter also receives. The adapter orchestrates tear-down of the ephemeral volume provisioned for this bucket.
+###### 3. COSI calls driver to generate credentials
 
-General prep:
-+ admin creates the needed bucket classes and bucket access classes.
+All but the sidecar for the specified provisioner will ignore the BucketClaim object. Only the appropriate sidecar will make a gRPC request to its driver to generate credentials/map service accounts.
 
-Prep for brownfield:
-+ admin creates `Bucket` instances representing backend buckets.
+This step will only proceed if the Bucket already exist. The Bucket's `BucketReady` condition should be true. Until access been granted, the `AccessGranted` condition in BucketAccess will be false.
 
-## Create Bucket
+```
+    COSI Driver (s3.amazonaws.com)
+    |------------------------------------------|
+    | grpc ProvisionerGrantBucketAccess({      |
+    |      "name": "ba-$uuid",                 |
+    |      "bucketID": "br-$uuid",             |
```

Comment:
Are all the occurrences of "br-$uuid" referring to a past version of this kep where we defined BucketRequests - thus _br_?